yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -313,6 +320,21 @@ private[spark] class Client(
     env("SPARK_YARN_MODE") = "true"
     env("SPARK_YARN_STAGING_DIR") = stagingDir
     env("SPARK_USER") = UserGroupInformation.getCurrentUser().getShortUserName()
+    // If we logged in from keytab, make sure we copy the keytab to the staging directory on
```

Comment:
this comment should probably be more like if we passed in a keytab. Whether we used it to login or not before submitted is irrelevant.